scalableminds/webknossos
# Use react error boundary to gracefully catch rendering bugs

At the moment, an error within a react component will simply crash wk. When developing, the page instantly refreshes without a chance to see the actual error (when dev tools are closed). If the error re-occurs on render, this causes an endless loop.

This PR fixes that issue by using a react error boundary, which catches rendering bugs and shows a toast message.

### Steps to test:
- ensure that normal rendering is unaffected
- introduce an error somewhere in a component (throw new Error in a render() method) and see that a toast is displayed



------
